pkg/labels/cidr/cidr.go

```diff
@@ -18,7 +22,7 @@ import (
 //
 // For IPv6 addresses, it converts ":" into "-" as EndpointSelectors don't
 // support colons inside the name section of a label.
-func maskedIPToLabelString(ip netip.Addr, prefix int) string {
+func maskedIPToLabel(ip netip.Addr, prefix int) labels.Label {
 	ipStr := ip.String()
 	ipNoColons := strings.Replace(ipStr, ":", "-", -1)
```

Comment:
I wonder if we can skip another allocation here by doing the character replacement in-place? (Strings are immutable but `[]byte` isn't... I wonder if the compiler is clever enough here).

Reply:
Good point! I think I found a way to avoid the allocation. I'm gonna open a follow up PR for that.